### PR TITLE
add link to reference doc

### PR DESCRIPTION
**Pull request by Netyyyy**
As title.linked https://github.com/Azure/azure-sdk-for-java/issues/27227

### CODE REVIEW

File: eventhubs/spring-cloud-azure-stream-binder-eventhubs/eventhubs-binder/README.md [hunk 1]
```diff
@@ -182,3 +182,8 @@ terraform -chdir=./terraform destroy -auto-approve
 ```shell
 terraform -chdir=terraform destroy -auto-approve
 ```
+## Enhancement
+###[Enable sync message](https://microsoft.github.io/spring-cloud-azure/4.0.0-beta.4/4.0.0-beta.4/reference/html/index.html#producer-properties)
```
Comment (backwind1233) [inline eventhubs/spring-cloud-azure-stream-binder-eventhubs/eventhubs-binder/README.md:186]: image
check format here.

File: servicebus/spring-cloud-azure-starter-integration-servicebus/single-namespace/README.md [hunk 2]
```diff
@@ -179,7 +179,6 @@ terraform -chdir=./terraform destroy -auto-approve
 ```shell
 terraform -chdir=terraform destroy -auto-approve
 ```
+## Enhancement
 
-
-
-
+[Set Service Bus message headers](https://microsoft.github.io/spring-cloud-azure/4.0.0-beta.4/4.0.0-beta.4/reference/html/index.html#si-sb-headers)
```
Comment (yiliuTo) [inline servicebus/spring-cloud-azure-starter-integration-servicebus/single-namespace/README.md:184]: Let's add some simple introduction here.

File: servicebus/spring-cloud-azure-stream-binder-servicebus/servicebus-multibinders/README.md [hunk 3]
```diff
@@ -187,3 +187,13 @@ terraform -chdir=./terraform destroy -auto-approve
 ```shell
 terraform -chdir=terraform destroy -auto-approve
 ```
+
+## Enhancement
+
```
Comment (yiliuTo) [inline servicebus/spring-cloud-azure-stream-binder-servicebus/servicebus-multibinders/README.md:192]: Let's add features like the way you update event hub binder readme, and the features should include: sync, resource provisioning, parititioning, session, error channels, message headers. And due to we don't show how to work in sync mode in this sample, thus we could only add text introduction without the code.

File: servicebus/spring-cloud-azure-stream-binder-servicebus/servicebus-queue-binder/README.md [hunk 4]
```diff
@@ -199,3 +199,13 @@ If you don't want to configure the connection string in your application, it's a
...
+
+## Enhancement
+
+### Configuration Options
+
+#### [Service Bus Producer Properties](https://microsoft.github.io/spring-cloud-azure/4.0.0-beta.4/4.0.0-beta.4/reference/html/index.html#producer-properties-2)
+
+#### [Service Bus Consumer Properties](https://microsoft.github.io/spring-cloud-azure/4.0.0-beta.4/4.0.0-beta.4/reference/html/index.html#consumer-properties)
+
+### [Set Service Bus message headers](https://microsoft.github.io/spring-cloud-azure/4.0.0-beta.4/4.0.0-beta.4/reference/html/index.html#scs-sb-headers)
```
Comment (yiliuTo) [inline servicebus/spring-cloud-azure-stream-binder-servicebus/servicebus-queue-binder/README.md:211]: same here.

File: servicebus/spring-cloud-azure-stream-binder-servicebus/servicebus-topic-binder/README.md [hunk 5]
```diff
@@ -197,3 +197,13 @@ If you don't want to configure the connection string in your application, it's a
 3. [Verify This Sample](#verify-this-sample).
 
 [application-rm.yaml]: https://github.com/Azure-Samples/azure-spring-boot-samples/blob/spring-cloud-azure_4.0.0-beta.4/servicebus/spring-cloud-azure-stream-binder-servicebus/servicebus-topic-binder/src/main/resources/application-rm.yaml
+
+## Enhancement
+
+### Configuration Options
+
+#### [Service Bus Producer Properties](https://microsoft.github.io/spring-cloud-azure/4.0.0-beta.4/4.0.0-beta.4/reference/html/index.html#producer-properties-2)
```
Comment (yiliuTo) [inline servicebus/spring-cloud-azure-stream-binder-servicebus/servicebus-topic-binder/README.md:205]: same here

File: servicebus/spring-cloud-azure-stream-binder-servicebus/servicebus-multibinders/README.md (hunk 3 above)
Comment (yiliuTo) [reply servicebus/spring-cloud-azure-stream-binder-servicebus/servicebus-multibinders/README.md:192]: Need we also udpate the event hub muti-binder sample?

**Review by yiliuTo — CHANGES_REQUESTED** [review]

File: eventhubs/spring-cloud-azure-stream-binder-eventhubs/eventhubs-binder/README.md [hunk 6]
```diff
@@ -182,3 +182,12 @@ terraform -chdir=./terraform destroy -auto-approve
...
 ```
+## Enhancement
+
+### [Enable sync message](https://microsoft.github.io/spring-cloud-azure/4.0.0-beta.4/4.0.0-beta.4/reference/html/index.html#producer-properties)
+
+To enable message sending in a synchronized way with Spring Cloud Stream 3.x, azure-spring-cloud-stream-binder-eventhubs supports the sync producer mode to get responses for sent messages. By enabling following configuration, you could use [StreamBridge](https://docs.spring.io/spring-cloud-stream/docs/3.1.3/reference/html/spring-cloud-stream.html#_sending_arbitrary_data_to_an_output_e_g_foreign_event_driven_sources) for the synchronized message producing.
+
+### [Using Batch Consuming](https://microsoft.github.io/spring-cloud-azure/4.0.0-beta.4/4.0.0-beta.4/reference/html/index.html#batch-consumer-support)
+
+[usage sample](https://microsoft.github.io/spring-cloud-azure/4.0.0-beta.4/4.0.0-beta.4/reference/html/index.html#batch-consumer-support-2)
```
[the remark this answers is not in the document]
Comment (Netyyyy) [reply eventhubs/spring-cloud-azure-stream-binder-eventhubs/eventhubs-binder/README.md:193]: add resource provision on @https://github.com/microsoft/spring-cloud-azure/pull/857

[the remark this answers is not in the document]
Comment (Netyyyy) [reply eventhubs/spring-cloud-azure-stream-binder-eventhubs/eventhubs-binder/README.md:193]: > We should introduce anther feature of resource provisioning, which is similar with https://github.com/Azure-Samples/azure-spring-boot-samples/tree/main/eventhubs/azure-spring-cloud-stream-binder-eventhubs/eventhubs-binder#enable-auto-create but we only support provisioning resources of event hub and consumer group.
> 
> And the necessary configuration are :
> 
> ```
> spring:
>   cloud:
>     azure:
>       credential:
>         tenant-id: tenant
>       profile:
>         subscription-id: id
>       eventhubs:
>         resource:
>           resource-group: group
> ```
> 
> Please also help to add such content in the reference doc, thanks.

add resource provision on @https://github.com/microsoft/spring-cloud-azure/pull/857

File: servicebus/spring-cloud-azure-starter-integration-servicebus/single-namespace/README.md [hunk 7]
```diff
@@ -179,7 +179,8 @@ terraform -chdir=./terraform destroy -auto-approve
 ```shell
 terraform -chdir=terraform destroy -auto-approve
 ```
+## Enhancement
 
+### Set Service Bus message headers
 
-
-
+users can get all the supported ServiceBus message headers [here](https://microsoft.github.io/spring-cloud-azure/4.0.0-beta.4/4.0.0-beta.4/reference/html/index.html#si-sb-headers) to configure.
```
Comment (yiliuTo) [inline servicebus/spring-cloud-azure-starter-integration-servicebus/single-namespace/README.md:186]: Should be "Users" instead "users"

File: servicebus/spring-cloud-azure-stream-binder-servicebus/servicebus-multibinders/README.md [hunk 8]
```diff
@@ -187,3 +187,36 @@ terraform -chdir=./terraform destroy -auto-approve
...
+### Partition Key Support
+
+ServiceBus binder supports partitioning by allowing setting partition key and session id in the message header. [Here](https://microsoft.github.io/spring-cloud-azure/4.0.0-beta.4/4.0.0-beta.4/reference/html/index.html#partition-key-support) shows how to set partition key for messages.
+
+### Session Support
+
+ServiceBus binder supports message sessions. [Here](https://microsoft.github.io/spring-cloud-azure/4.0.0-beta.4/4.0.0-beta.4/reference/html/index.html#session-support) shows how to set session id of a message.
+
+### ERROR CHANNEL
```
Comment (yiliuTo) [inline servicebus/spring-cloud-azure-stream-binder-servicebus/servicebus-multibinders/README.md:216]: same here

File: servicebus/spring-cloud-azure-stream-binder-servicebus/servicebus-multibinders/README.md [hunk 9]
```diff
@@ -187,3 +187,36 @@ terraform -chdir=./terraform destroy -auto-approve
...
+
+### Configuration Options
+
+[Service Bus Producer Properties](https://microsoft.github.io/spring-cloud-azure/4.0.0-beta.4/4.0.0-beta.4/reference/html/index.html#producer-properties-2)
+
+[Service Bus Consumer Properties](https://microsoft.github.io/spring-cloud-azure/4.0.0-beta.4/4.0.0-beta.4/reference/html/index.html#consumer-properties)
+
+### Resource Provision
+
+[Auto create resource](https://microsoft.github.io/spring-cloud-azure/4.0.0-beta.4/4.0.0-beta.4/reference/html/index.html#resource-provision-2): queue, topic, subscription
```
Comment (yiliuTo) [inline servicebus/spring-cloud-azure-stream-binder-servicebus/servicebus-multibinders/README.md:206]: same here, let's write a complete and formor sentence

File: servicebus/spring-cloud-azure-stream-binder-servicebus/servicebus-multibinders/README.md [hunk 10]
```diff
@@ -187,3 +187,36 @@ terraform -chdir=./terraform destroy -auto-approve
...
+
+### Partition Key Support
+
+ServiceBus binder supports partitioning by allowing setting partition key and session id in the message header. [Here](https://microsoft.github.io/spring-cloud-azure/4.0.0-beta.4/4.0.0-beta.4/reference/html/index.html#partition-key-support) shows how to set partition key for messages.
```
Comment (yiliuTo) [inline servicebus/spring-cloud-azure-stream-binder-servicebus/servicebus-multibinders/README.md:210]: Should be "Service Bus"

File: servicebus/spring-cloud-azure-stream-binder-servicebus/servicebus-multibinders/README.md [hunk 11]
```diff
@@ -187,3 +187,36 @@ terraform -chdir=./terraform destroy -auto-approve
...
+
+### Session Support
+
+ServiceBus binder supports message sessions. [Here](https://microsoft.github.io/spring-cloud-azure/4.0.0-beta.4/4.0.0-beta.4/reference/html/index.html#session-support) shows how to set session id of a message.
+
+### ERROR CHANNEL
+
+ServiceBus binder supports consumer error channel, producer error channel and global default error channel, click [here](https://microsoft.github.io/spring-cloud-azure/4.0.0-beta.4/4.0.0-beta.4/reference/html/index.html#error-channels-2)  to see more information.
```
Comment (yiliuTo) [inline servicebus/spring-cloud-azure-stream-binder-servicebus/servicebus-multibinders/README.md:218]: same here

File: servicebus/spring-cloud-azure-stream-binder-servicebus/servicebus-queue-binder/README.md [hunk 12]
```diff
@@ -199,3 +199,36 @@ If you don't want to configure the connection string in your application, it's a
...
+
+### Configuration Options
+
+[Service Bus Producer Properties](https://microsoft.github.io/spring-cloud-azure/4.0.0-beta.4/4.0.0-beta.4/reference/html/index.html#producer-properties-2)
+
+[Service Bus Consumer Properties](https://microsoft.github.io/spring-cloud-azure/4.0.0-beta.4/4.0.0-beta.4/reference/html/index.html#consumer-properties)
+
+### Resource Provision
+
+[Auto create resource](https://microsoft.github.io/spring-cloud-azure/4.0.0-beta.4/4.0.0-beta.4/reference/html/index.html#resource-provision-2): queue, topic, subscription
```
Comment (yiliuTo) [inline servicebus/spring-cloud-azure-stream-binder-servicebus/servicebus-queue-binder/README.md:218]: same here.

File: servicebus/spring-cloud-azure-stream-binder-servicebus/servicebus-queue-binder/README.md [hunk 13]
```diff
@@ -199,3 +199,36 @@ If you don't want to configure the connection string in your application, it's a
...
+
+### Partition Key Support
+
+ServiceBus binder supports partitioning by allowing setting partition key and session id in the message header. [Here](https://microsoft.github.io/spring-cloud-azure/4.0.0-beta.4/4.0.0-beta.4/reference/html/index.html#partition-key-support) shows how to set partition key for messages.
+
+### Session Support
+
```
Comment (yiliuTo) [inline servicebus/spring-cloud-azure-stream-binder-servicebus/servicebus-queue-binder/README.md:225]: same here

File: servicebus/spring-cloud-azure-stream-binder-servicebus/servicebus-queue-binder/README.md [hunk 14]
```diff
@@ -199,3 +199,36 @@ If you don't want to configure the connection string in your application, it's a
...
+ServiceBus binder supports message sessions. [Here](https://microsoft.github.io/spring-cloud-azure/4.0.0-beta.4/4.0.0-beta.4/reference/html/index.html#session-support) shows how to set session id of a message.
+
+### ERROR CHANNEL
+
+ServiceBus binder supports consumer error channel, producer error channel and global default error channel, click [here](https://microsoft.github.io/spring-cloud-azure/4.0.0-beta.4/4.0.0-beta.4/reference/html/index.html#error-channels-2)  to see more information.
```
Comment (yiliuTo) [inline servicebus/spring-cloud-azure-stream-binder-servicebus/servicebus-queue-binder/README.md:230]: same here

File: servicebus/spring-cloud-azure-stream-binder-servicebus/servicebus-topic-binder/README.md [hunk 15]
```diff
@@ -197,3 +197,36 @@ If you don't want to configure the connection string in your application, it's a
...
+
+[Service Bus Consumer Properties](https://microsoft.github.io/spring-cloud-azure/4.0.0-beta.4/4.0.0-beta.4/reference/html/index.html#consumer-properties)
+
+### Resource Provision
+
+[Auto create resource](https://microsoft.github.io/spring-cloud-azure/4.0.0-beta.4/4.0.0-beta.4/reference/html/index.html#resource-provision-2): queue, topic, subscription
+
+### Partition Key Support
+
+ServiceBus binder supports partitioning by allowing setting partition key and session id in the message header. [Here](https://microsoft.github.io/spring-cloud-azure/4.0.0-beta.4/4.0.0-beta.4/reference/html/index.html#partition-key-support) shows how to set partition key for messages.
```
Comment (yiliuTo) [inline servicebus/spring-cloud-azure-stream-binder-servicebus/servicebus-topic-binder/README.md:220]: same here

File: servicebus/spring-cloud-azure-stream-binder-servicebus/servicebus-topic-binder/README.md [hunk 16]
```diff
@@ -197,3 +197,36 @@ If you don't want to configure the connection string in your application, it's a
...
+
+### Session Support
+
+ServiceBus binder supports message sessions. [Here](https://microsoft.github.io/spring-cloud-azure/4.0.0-beta.4/4.0.0-beta.4/reference/html/index.html#session-support) shows how to set session id of a message.
```
Comment (yiliuTo) [inline servicebus/spring-cloud-azure-stream-binder-servicebus/servicebus-topic-binder/README.md:224]: same here

File: servicebus/spring-cloud-azure-stream-binder-servicebus/servicebus-topic-binder/README.md [hunk 17]
```diff
@@ -197,3 +197,36 @@ If you don't want to configure the connection string in your application, it's a
...
+
+### ERROR CHANNEL
```
Comment (yiliuTo) [inline servicebus/spring-cloud-azure-stream-binder-servicebus/servicebus-topic-binder/README.md:226]: same here and below

File: servicebus/spring-cloud-azure-stream-binder-servicebus/servicebus-queue-binder/README.md [hunk 18]
```diff
@@ -199,3 +199,36 @@ If you don't want to configure the connection string in your application, it's a
...
+
+### Partition Key Support
+
+ServiceBus binder supports partitioning by allowing setting partition key and session id in the message header. [Here](https://microsoft.github.io/spring-cloud-azure/4.0.0-beta.4/4.0.0-beta.4/reference/html/index.html#partition-key-support) shows how to set partition key for messages.
+
+### Session Support
+
+ServiceBus binder supports message sessions. [Here](https://microsoft.github.io/spring-cloud-azure/4.0.0-beta.4/4.0.0-beta.4/reference/html/index.html#session-support) shows how to set session id of a message.
+
+### ERROR CHANNEL
```
Comment (yiliuTo) [inline servicebus/spring-cloud-azure-stream-binder-servicebus/servicebus-queue-binder/README.md:228]: same here

**Review by yiliuTo — CHANGES_REQUESTED** [review]

File: eventhubs/spring-cloud-azure-stream-binder-eventhubs/eventhubs-binder/README.md [hunk 19]
```diff
@@ -182,3 +182,88 @@ terraform -chdir=./terraform destroy -auto-approve
...
+### Resource Provision
+
+Event Hubs binder supports provisioning of event hub and consumer group, users could use [properties](https://microsoft.github.io/spring-cloud-azure/4.0.0-beta.4/4.0.0-beta.4/reference/html/index.html#resource-provision) to enable provisioning.
+
+### Partitioning Support
+
+A PartitionSupplier with user-provided partition information will be created to configure the partition information about the message to be sent, [Here](https://microsoft.github.io/spring-cloud-azure/4.0.0-beta.4/4.0.0-beta.4/reference/html/index.html#partitioning-support-2) is the process of obtaining different priorities of the partition ID and key.
+
+#### Partition key support
+The binder supports Event Hubs partitioning by allowing setting partition key and session id in the message header. This section introduces how to set partition key for messages.
```
Comment (yiliuTo) [inline eventhubs/spring-cloud-azure-stream-binder-eventhubs/eventhubs-binder/README.md:239]: Sorry there is wrong information I pass to you. There are some differences for partitioning support between Event Hubs and Service Bus. Event Hubs only support partition key and id, so there is no session id for eh. And the way to configure partition information is also different from sb, so let's change these several parts:
1. Change the title from "Partition key support" to "Partition support".
2. Delete below content and code, only remain "The binder supports Event Hubs partitioning by allowing setting partition key and id. Please refer to the [reference doc](https://microsoft.github.io/spring-cloud-azure/4.0.0-beta.4/4.0.0-beta.4/reference/html/index.html#partitioning-support-2) for more details."
3. Remove below section of "Session support"

File: eventhubs/spring-cloud-azure-stream-binder-eventhubs/eventhubs-multibinders/README.md [hunk 20]
```diff
@@ -185,6 +185,61 @@ terraform -chdir=./terraform destroy -auto-approve
...
+
+### Resource Provision
+
+Event Hubs binder supports provisioning of event hub and consumer group, users could use [properties](https://microsoft.github.io/spring-cloud-azure/4.0.0-beta.4/4.0.0-beta.4/reference/html/index.html#resource-provision) to enable provisioning.
+
+### Partitioning Support
+
+A PartitionSupplier with user-provided partition information will be created to configure the partition information about the message to be sent, [Here](https://microsoft.github.io/spring-cloud-azure/4.0.0-beta.4/4.0.0-beta.4/reference/html/index.html#partitioning-support-2) is the process of obtaining different priorities of the partition ID and key.
+
+#### Partition key support
```
Comment (yiliuTo) [inline eventhubs/spring-cloud-azure-stream-binder-eventhubs/eventhubs-multibinders/README.md:214]: Let's remove the whole section of "Partition key support"

File: eventhubs/spring-cloud-azure-stream-binder-eventhubs/eventhubs-multibinders/README.md [hunk 21]
```diff
@@ -185,6 +185,61 @@ terraform -chdir=./terraform destroy -auto-approve
...
+    return () -> {
+        String value = “random payload”;
+        return MessageBuilder.withPayload(value)
+            .setHeader("<message-header-key>", value.length() % 4)
+            .build();
+    };
+}
+```
+
+#### Session support
```
Comment (yiliuTo) [inline eventhubs/spring-cloud-azure-stream-binder-eventhubs/eventhubs-multibinders/README.md:230]: same here, remove this section.

**Review by yiliuTo — CHANGES_REQUESTED** [review]

**Review by yiliuTo — APPROVED** [review]
LGTM

**Pull request merged**